src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowDelete.cs

```diff
@@ -65,15 +65,42 @@ public override DbCommand GetCommand(DbConnection connection)
 
             // Return a SqlCommand with formatted with the parameters from the where clause
             WhereClause where = GetWhereClause(true);
-            string commandText = GetCommandText(where.CommandText);
-
             DbCommand command = connection.CreateCommand();
+            string commandText = "";
+            if (!this.checkIfValid(connection, where))
+            {
+                commandText = " DECLARE @error NVARCHAR(100) = N'INVALID DELETE: rows with duplicate or text values are not currently supported';" + Environment.NewLine +
+                                                           " RAISERROR (@error, 16, 1)";
+            }
+            else
+            {
+                commandText = GetCommandText(where.CommandText);
+            }
+
             command.CommandText = commandText;
             command.Parameters.AddRange(where.Parameters.ToArray());
-
             return command;
         }
 
+        private Boolean checkIfValid(DbConnection connection, WhereClause where)
+        {
+            string formatString = "SELECT COUNT(*) FROM {0} {1}";
```

Comment:
String.format could lead to a sql injection, you should use SqlParameters.

You can find examples of them https://github.com/microsoft/sqltoolsservice/blob/427f0c11e8d0c1db0a81f4c6023215001dd0e32d/src/Microsoft.SqlTools.ServiceLayer/Agent/Jobs/AgentNotebookHelper.cs